From d0cec2ecd9fe0f1fe5f7e292850784109c04daaf Mon Sep 17 00:00:00 2001
From: qi duan <qi.duan@amlogic.com>
Date: Tue, 2 Jul 2019 06:31:26 -0400
Subject: [PATCH] dts: usb/pci: Setup S400 usb and pci with upstream
 kernel.[1/1]

PD#SWPL-10529

Problem:
need support usb and pci on S400

Solution:
enable usb and pci dts and config

Verify:
s400

Change-Id: Ib3e74755dc0c0c408782a09d6e44a0f3ea3b0237
Signed-off-by: qi duan <qi.duan@amlogic.com>
---
 arch/arm64/boot/dts/amlogic/meson-axg.dtsi    | 116 ++++++++++++++++++
 arch/arm64/configs/defconfig                  |   1 +
 3 files changed, 131 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
index eaf066c7128c..3860f63e11fd 100644
--- a/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-axg.dtsi
@@ -1546,6 +1546,28 @@
 			#address-cells = <0>;
 		};
 
+		usb0: usb@ff500000 {
+			compatible = "amlogic,meson-axg-dwc3";
+			#address-cells = <2>;
+			#size-cells = <2>;
+			ranges;
+
+			clocks = <&clkc CLKID_USB>;
+			clock-names = "usb_general";
+			resets = <&reset RESET_USB_OTG>;
+			reset-names = "usb_otg";
+
+			dwc3: dwc3@ff500000 {
+				compatible = "snps,dwc3";
+				reg = <0x0 0xff500000 0x0 0x100000>;
+				interrupts = <GIC_SPI 30 IRQ_TYPE_LEVEL_HIGH>;
+				dr_mode = "host";
+				maximum-speed = "high-speed";
+					snps,dis_u2_susphy_quirk;
+				phys = <&usb2_phy1>, <&usb3_phy>;
+			};
+		};
+
 		cbus: bus@ffd00000 {
 			compatible = "simple-bus";
 			reg = <0x0 0xffd00000 0x0 0x25000>;
@@ -1704,8 +1726,102 @@
 				clock-names = "core", "clkin0", "clkin1";
 				resets = <&reset RESET_SD_EMMC_C>;
 			};
+
+			usb2_phy1: phy@9020 {
+				compatible = "amlogic,meson-gxl-usb2-phy";
+				#phy-cells = <0>;
+				reg = <0x0 0x9020 0x0 0x20>;
+				resets = <&reset RESET_USB_OTG>;
+				reset-names = "phy";
+				status = "okay";
+			};
+
+			usb3_phy: phy@9080 {
+				compatible = "amlogic,meson-gxl-usb3-phy";
+				#phy-cells = <0>;
+				reg = <0x0 0x9080 0x0 0x20>;
+				resets = <&reset RESET_USB_OTG>, <&reset RESET_USB_OTG>;
+				reset-names = "phy", "peripheral";
+				clocks = <&clkc CLKID_USB>, <&clkc CLKID_CLK81>;
+				clock-names = "phy", "peripheral";
+				phys = <&usb2_phy1>;
+				phy-names = "usb2-phy";
+				status = "okay";
+			};
 		};
 
+
+	pcie_A: pcie@f9800000 {
+		compatible = "amlogic,axg-pcie", "snps,dw-pcie";
+		reg = <0x0 0xf9800000 0x0 0x400000
+			0x0 0xff646000 0x0 0x2000
+			0x0 0xff644000 0x0 0x2000
+			0x0 0xf9f00000 0x0 0x100000>;
+		reg-names = "elbi", "cfg", "phy", "config";
+		interrupts = <GIC_SPI 177 IRQ_TYPE_EDGE_RISING>;
+		#interrupt-cells = <1>;
+		interrupt-map-mask = <0 0 0 0>;
+		interrupt-map = <0 0 0 0 &gic GIC_SPI 179 IRQ_TYPE_EDGE_RISING>;
+		bus-range = <0x0 0xff>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		device_type = "pci";
+		ranges = <0x82000000 0 0 0x0 0xf9c00000 0 0x00300000>;
+		num-lanes = <1>;
+		clocks = <&clkc CLKID_USB
+			&clkc CLKID_MIPI_ENABLE
+			&clkc CLKID_PCIE_A
+			&clkc CLKID_PCIE_CML_EN0>;
+		clock-names = "pcie_general",
+				"pcie_mipi_en",
+				"pcie",
+				"port";
+		resets = <&reset RESET_PCIE_PHY>,
+			<&reset RESET_PCIE_A>,
+			<&reset RESET_PCIE_APB>;
+		reset-names = "phy",
+				"port",
+				"apb";
+		status = "disable";
+	};
+
+	pcie_B: pcie@fa000000 {
+		compatible = "amlogic,axg-pcie", "snps,dw-pcie";
+		reg = <0x0 0xfa000000 0x0 0x400000
+			0x0 0xff648000 0x0 0x2000
+			0x0 0xff644000 0x0 0x2000
+			0x0 0xfa400000 0x0 0x100000>;
+		reg-names = "elbi", "cfg", "phy", "config";
+		interrupts = <GIC_SPI 167 IRQ_TYPE_EDGE_RISING>;
+		#interrupt-cells = <1>;
+		bus-range = <0x0 0xff>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		interrupt-map-mask = <0 0 0 0>;
+		interrupt-map = <0 0 0 0 &gic GIC_SPI 169 IRQ_TYPE_EDGE_RISING>;
+		device_type = "pci";
+		ranges = <0x81000000 0 0 0 0xfa500000 0x0 0x10000
+			/* downstream I/O */
+			0x82000000 0 0xfa510000 0x0 0xfa510000 0 0x002f0000>;
+			/* non-prefetchable memory */
+		num-lanes = <1>;
+
+		clocks = <&clkc CLKID_USB
+			&clkc CLKID_MIPI_ENABLE
+			&clkc CLKID_PCIE_B
+			&clkc CLKID_PCIE_CML_EN1>;
+		clock-names = "pcie_general",
+				"pcie_mipi_en",
+				"pcie",
+				"port";
+		resets = <&reset RESET_PCIE_PHY>,
+			<&reset RESET_PCIE_B>,
+			<&reset RESET_PCIE_APB>;
+		reset-names = "phy",
+				"port",
+				"apb";
+		status = "disable";
+	};
 		sram: sram@fffc0000 {
 			compatible = "amlogic,meson-axg-sram", "mmio-sram";
 			reg = <0x0 0xfffc0000 0x0 0x20000>;
-- 
2.26.2

